scipy/stats/_binomtest.py

```diff
@@ -0,0 +1,258 @@
+
+    Returns proportion, prop_low, prop_high
+    """
+    if alternative == 'two-sided':
+        alpha = (1 - confidence_level) / 2
+        if k == 0:
+            plow = 0.0
+        else:
+            plow = brentq(lambda p: binom.sf(k-1, n, p) - alpha, 0, 1)
+        if k == n:
+            phigh = 1.0
+        else:
+            phigh = brentq(lambda p: binom.cdf(k, n, p) - alpha, 0, 1)
+    elif alternative == 'less':
+        alpha = 1 - confidence_level
+        plow = 0.0
+        if k == n:
+            phigh = 1.0
+        else:
+            phigh = brentq(lambda p: binom.cdf(k, n, p) - alpha, 0, 1)
+    elif alternative == 'greater':
+        alpha = 1 - confidence_level
+        if k == 0:
+            plow = 0.0
+        else:
+            plow = brentq(lambda p: binom.sf(k-1, n, p) - alpha, 0, 1)
+        phigh = 1.0
+    return plow, phigh
+
+
+def _binom_wilson_conf_int(k, n, confidence_level, alternative, correction):
+    # This function assumes that the arguments have already been validated.
+    # In particular, `alternative` must be one of 'two-sided', 'less' or
+    # 'greater'.
+    p = k / n
+    if alternative == 'two-sided':
+        z = ndtri(0.5 + 0.5*confidence_level)
+    else:
+        z = ndtri(confidence_level)
+
+    t = 1 + z**2/n
+    r = (p + z**2/(2*n)) / t
+
+    if correction:
+        if alternative == 'less' or k == 0:
+            lo = 0.0
+        else:
+            dlo = ((z * sqrt(z**2 - 1/n + 4*n*p*(1 - p) + (4*p - 2)) + 1) /
+                   (2*n*t))
+            lo = r - dlo
+        if alternative == 'greater' or k == n:
+            hi = 1.0
+        else:
+            dhi = ((z * sqrt(z**2 - 1/n + 4*n*p*(1 - p) - (4*p - 2)) + 1) /
+                   (2*n*t))
+            hi = r + dhi
+    else:
+        d = z/t * sqrt(p*(1-p)/n + (z/(2*n))**2)
+        if alternative == 'less' or k == 0:
+            lo = 0.0
+        else:
+            lo = r - d
+        if alternative == 'greater' or k == n:
+            hi = 1.0
+        else:
+            hi = r + d
+
+    return lo, hi
+
+
+def binomtest(k, *, n=None, p=0.5, alternative='two-sided'):
+    """
+    Perform a test that the probability of success is p.
+
+    This is a test of the null hypothesis that the probability of success
+    in a Bernoulli experiment is `p`.
+
+    Parameters
+    ----------
+    k : int or array_like
+        The number of successes, or if k has length 2, it is the
+        number of successes and the number of failures.
+    n : int
+        The number of trials.  This is ignored if k gives both the
+        number of successes and failures.
+    p : float, optional
+        The hypothesized probability of success.  ``0 <= p <= 1``. The
+        default value is ``p = 0.5``.
+    alternative : {'two-sided', 'greater', 'less'}, optional
+        Indicates the alternative hypothesis. The default value is
+        'two-sided'.
+
+    Returns
+    -------
+    result : BinomTestResult instance
```

Comment:
Even if we have a generic `StatsTestResult`, the various tests should probably return subclasses of that class, because they will likely have attributes and methods specific to the test.  I'm not completely opposed to a base class, but at the moment I'm not convinced we need it.  I'll spend some more time today looking into it.